### PR TITLE
Create multiple clearsky endpoint 

### PR DESCRIPTION
**Pull request by ericcccsliu**
# Pull Request

## Description
For our UI, we occasionally need to get multiple clearsky forecasts at once. This pull request adds an endpoint to get clearsky estimates over many sites, which also brings the clearsky endpoints in line with the endpoints for forecasts and actuals

## How Has This Been Tested?

- [x] Added tests 
- [x] Performed a sanity check with outputted data 

The data processing shouldn't be changed at all—doing the same processing, just with multiple sites 

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pv_site_api/main.py [hunk 1]
```diff
@@ -319,53 +320,68 @@ def get_pv_estimate_clearsky(site_uuid: str, session: Session = Depends(get_sess
     """
     ### Gets a estimate of AC production under a clear sky
     """
+    clearsky_estimates = get_pv_estimate_clearsky_many_sites(site_uuid, session)
```
Comment (AndrewLester) [inline pv_site_api/main.py:323]: Should we add 404 logic like they have above?

https://github.com/openclimatefix/pv-site-api/pull/84/files#diff-aa1f06de99546f36c9596d3097374fac5b2c99ac6cae3d4cce042440bcfe46f6R274-R280

File: pyproject.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ pydantic = "^1.10.5"
 uvicorn = {extras = ["standard"], version = "^0.20.0"}
 psycopg2-binary = "^2.9.5"
 sqlalchemy = "^1.4.46"
-pvsite-datamodel = "^0.1.30"
+pvsite-datamodel = "0.1.30"
```
Comment (AndrewLester) [inline pyproject.toml:15]: Is there a reason for removing this?

File: tests/test_clearsky.py [hunk 3]
```diff
@@ -11,10 +11,31 @@ def test_get_clearsky_fake(client, fake):
...
+def test_get_clearsky(db_session, client, sites):
+    site_uuid = sites[0].site_uuid
     response = client.get(f"/sites/{site_uuid}/clearsky_estimate")
     assert response.status_code == 200
 
     clearsky_estimate = ClearskyEstimate(**response.json())
     assert len(clearsky_estimate.clearsky_estimate) > 0
+
+
+def test_get_forecast_many_sites(db_session, client, sites):
```
Comment (AndrewLester) [inline tests/test_clearsky.py:32]: Should this be?
```suggestion
def test_get_clearsky_many_sites(db_session, client, sites):
```

**Review by AndrewLester — APPROVED** [review]

File: pv_site_api/main.py (hunk 1 above)
Comment (ericcccsliu) [reply pv_site_api/main.py:328]: yep, done!

File: pyproject.toml (hunk 2 above)
Comment (ericcccsliu) [reply pyproject.toml:15]: Nope, accidental, putting it back

File: tests/test_clearsky.py (hunk 3 above)
Comment (ericcccsliu) [reply tests/test_clearsky.py:32]: yep, thanks

**Comment by ericcccsliu** [comment]
For some reason, a check regarding the forecast is failing (length of forecast values is 17 instead of 20) and I can't seem to figure out why

File: pv_site_api/main.py [hunk 4]
```diff
@@ -319,53 +320,68 @@ def get_pv_estimate_clearsky(site_uuid: str, session: Session = Depends(get_sess
...
+        loc = location.Location(site.latitude, site.longitude)
+
+        # Create DatetimeIndex over four days, with a frequency of 15 minutes.
+        # Starts from midnight yesterday.
+        times = pd.date_range(start=get_yesterday_midnight(), periods=384, freq="15min", tz="UTC")
+        clearsky = loc.get_clearsky(times)
+        solar_position = loc.get_solarposition(times=times)
+
+        # Using default tilt of 0 and orientation of 180 from defaults of PVSystem
+        tilt = site.tilt if site.tilt is not None else 0
```
Comment (simlmx) [inline pv_site_api/main.py:354]:
```suggestion
        tilt = site.tilt or 0
```

It's also good the way you wrote it (more semantic!), just pointing out that this is possible in python! The sme would work for `orientation`.

**Review by simlmx — APPROVED** [review]
LGTM

File: pv_site_api/main.py (hunk 4 above)
Comment (ericcccsliu) [reply pv_site_api/main.py]: Oh thanks, I didn't know this was possible! Changed it to your suggestion—think it's more clear

File: pv_site_api/main.py (hunk 1 above)
Comment (peterdudfield) [reply pv_site_api/main.py:323]: I always get a bit worried when we take stuff. Should this be put back in as @AndrewLester suggests

**Comment by peterdudfield** [comment]
@simlmx would you mind deploying this?

**Pull request merged**